Ingester crash during chunk transfer

```
ts=2018-03-27T14:09:56.964261909Z caller=log.go:112 level=info msg="changing ingester state from" old_state=JOINING new_state=PENDING
panic: runtime error: index out of range
goroutine 175 [running]:
github.com/weaveworks/cortex/pkg/ingester.(*memorySeries).setChunks(0xc49eb48440, 0x1816c98, 0x0, 0x0, 0xc4b429a4b0, 0xc17b0536b046547f)
/go/src/github.com/weaveworks/cortex/pkg/ingester/series.go:173 +0x10d
github.com/weaveworks/cortex/pkg/ingester.(*Ingester).TransferChunks(0xc42038ef00, 0x17b45c0, 0xc42004fcb0, 0x0, 0x0)
/go/src/github.com/weaveworks/cortex/pkg/ingester/ingester_claim.go:91 +0x3a5
github.com/weaveworks/cortex/pkg/ingester/client._Ingester_TransferChunks_Handler(0xf54260, 0xc42038ef00, 0x17b32a0, 0xc420088aa0, 0xc42058fc68, 0x79a47c)
/go/src/github.com/weaveworks/cortex/pkg/ingester/client/cortex.pb.go:1809 +0xb2
github.com/weaveworks/cortex/vendor/google.golang.org/grpc.(*Server).processStreamingRPC(0xc420340160, 0x17b4200, 0xc42017c780, 0xc4207eef00, 0xc42044a1b0, 0x178d5e0, 0x0, 0x0, 0x0)
/go/src/github.com/weaveworks/cortex/vendor/google.golang.org/grpc/server.go:1053 +0xa41
github.com/weaveworks/cortex/vendor/google.golang.org/grpc.(*Server).handleStream(0xc420340160, 0x17b4200, 0xc42017c780, 0xc4207eef00, 0x0)
/go/src/github.com/weaveworks/cortex/vendor/google.golang.org/grpc/server.go:1147 +0x14c1
github.com/weaveworks/cortex/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc420188be0, 0xc420340160, 0x17b4200, 0xc42017c780, 0xc4207eef00)
/go/src/github.com/weaveworks/cortex/vendor/google.golang.org/grpc/server.go:638 +0x9f
created by github.com/weaveworks/cortex/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
/go/src/github.com/weaveworks/cortex/vendor/google.golang.org/grpc/server.go:636 +0xa1
```

As far as I can see this means that an empty series was sent, so `[len(descs)-1` was negative.

--- a/pkg/ingester/ingester_lifecycle.go
+++ b/pkg/ingester/ingester_lifecycle.go
@@ -375,6 +375,11 @@ func (i *Ingester) transferChunks() error {
 		for pair := range state.fpToSeries.iter() {
 			state.fpLocker.Lock(pair.fp)
 
+			if len(pair.series.chunkDescs) == 0 { // Nothing to send?
+				state.fpLocker.Unlock(pair.fp)
+				continue
+			}
+
 			chunks, err := toWireChunks(pair.series.chunkDescs)
 			if err != nil {
 				state.fpLocker.Unlock(pair.fp)

--- a/pkg/ingester/series.go
+++ b/pkg/ingester/series.go
@@ -170,7 +170,9 @@ func (s *memorySeries) setChunks(descs []*desc) error {
 	}
 
 	s.chunkDescs = descs
-	s.lastTime = descs[len(descs)-1].LastTime
+	if len(descs) > 0 {
+		s.lastTime = descs[len(descs)-1].LastTime
+	}
 	return nil
 }
 

